src/build.d

```diff
@@ -0,0 +1,1131 @@
+        env["HOST_DMD_KIND"] = "gdc";
+    else
+        enforce(0, "Invalid Host DMD found: " ~ hostDMDVersion);
+
+    env.getDefault("ENABLE_WARNINGS", "0");
+    string[] warnings;
+    if (env["ENABLE_WARNINGS"] != "0")
+    {
+        warnings = ["-Wall", "-Wextra", "-Werror",
+            "-Wno-attributes",
+            "-Wno-char-subscripts",
+            "-Wno-deprecated",
+            "-Wno-empty-body",
+            "-Wno-format",
+            "-Wno-missing-braces",
+            "-Wno-missing-field-initializers",
+            "-Wno-overloaded-virtual",
+            "-Wno-parentheses",
+            "-Wno-reorder",
+            "-Wno-return-type",
+            "-Wno-sign-compare",
+            "-Wno-strict-aliasing",
+            "-Wno-switch",
+            "-Wno-type-limits",
+            "-Wno-unknown-pragmas",
+            "-Wno-unused-function",
+            "-Wno-unused-label",
+            "-Wno-unused-parameter",
+            "-Wno-unused-value",
+            "-Wno-unused-variable",
+        ];
+        if (env["CXX_KIND"] == "g++")
+            warnings ~= [
+                "-Wno-logical-op",
+                "-Wno-narrowing",
+                "-Wno-unused-but-set-variable",
+                "-Wno-uninitialized",
+                "-Wno-class-memaccess",
+                "-Wno-implicit-fallthrough",
+            ];
+        else if (env["CXX_KIND"] == "clang++")
+            warnings ~= "-Wno-logical-op-parentheses";
+    }
+    else
+    {
+        warnings = ["-Wno-deprecated", "-Wstrict-aliasing", "-Werror"];
+    }
+
+    auto targetCPU = "X86";
+    auto cxxFlags = warnings;
+    cxxFlags ~= [
+        "-fno-exceptions", "-fno-rtti",
+        "-D__pascal=", "-DMARS=1", "-DTARGET_"~os.toUpper~"=1",
+        "-DDM_TARGET_CPU_"~targetCPU~"=1",
+        env["MODEL_FLAG"],
+        env["PIC_FLAG"],
+    ];
+    if (env["CXX_KIND"] == "g++")
+        cxxFlags ~= ["-std=gnu++98"];
+    if (env["CXX_KIND"] == "clang++")
+        cxxFlags ~= ["-xc++"];
+
+    flags["CXXFLAGS"] = cxxFlags;
+
+    // TODO: allow adding new flags from the environment
+    string[] dflags;
+    flags["DFLAGS"] ~= ["-version=MARS", "-w", "-de", env["PIC_FLAG"], env["MODEL_FLAG"]];
+    // TODO: add different ENABLE modes
+
+    flags["BACK_FLAGS"] = ["-I"~env["ROOT"], "-I"~env["TK"], "-I"~env["C"], "-I"~env["G"], "-I"~env["D"], "-DDMDV2=1"];
+
+    // TODO: add support for dObjc
+    auto dObjc = false;
+    version(OSX) version(X86_64)
+        dObjc = true;
+
+    return env;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// D source files
+////////////////////////////////////////////////////////////////////////////////
+
+auto sourceFiles()
+{
+    struct Sources
+    {
+        // TODO: remove unused file lists
+        string[] frontend, lexer, lexerRoot, root, glue, dmd, backend;
+        string[] backendHeaders, tkHeaders, backendC, tkC, backendObjects;
```

Comment:
We could use some globbing here to find all the files which would probably cut down half of the size of this script.
The problem is that the frontend is separated in multiple parts and we would have to do a bit of filtering and thus maybe not save so much.